calculator.go

```diff
@@ -1,13 +1,91 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+// Add returns the sum of two or more numbers.
+func Add(a, b float64, v ...float64) float64 {
+	r := a + b
+	for _, x := range v {
+		r += x
+	}
+	return r
+}
+
+// Subtract returns the difference of two or more numbers.
+func Subtract(a, b float64, v ...float64) float64 {
+	r := a - b
+	for _, x := range v {
+		r -= x
+	}
+	return r
+}
+
+// Multiply returns the product of two or more numbers.
+func Multiply(a, b float64, v ...float64) float64 {
+	r := a * b
+	for _, x := range v {
+		r *= x
+	}
+	return r
+}
+
+// Divide returns the quotient of two or more numbers.
+func Divide(a, b float64, v ...float64) (float64, error) {
+	if b == 0 {
+		return 0.0, fmt.Errorf("divide-by-zero for Divide(%f, %f)", a, b)
+	}
+
+	r := a / b
+	for i, x := range v {
+		if x == 0 {
+			return 0.0, fmt.Errorf("divide-by-zero for Divide(%f, %f), in iteration %v processing variadic float64", r, x, i)
+		}
+		r /= x
+	}
+	return r, nil
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// Sqrt returns the square root of a number
+func Sqrt(a float64) (float64, error) {
+	if a == 0 {
+		return 0.0, fmt.Errorf("cannot-get-square-root-of-a-negative-number for Sqrt(%f)", a)
+	}
+
+	return math.Sqrt(a), nil
+}
+
+func EvaluateExpression(e string) (float64, error) {
+	var a, b float64
+	var operator string
+
+	strings.ReplaceAll(e, " ", "") // remove all spaces
+
+	numFields, err := fmt.Sscanf(e, "%f%1s%f", &a, &operator, &b)
+	if err != nil {
+		return 0.0, fmt.Errorf("Expression() error while parsing %q: %v", e, err)
+	}
+
+	if numFields < 3 {
```

Comment:
In fact, `numFields` needs to be exactly 3, doesn't it? Three shall be the number of fields, and the number of fields shall be exactly three. _Four_ shalt thou not count...